DeezyMatch/tests/test_utils.py

```diff
@@ -0,0 +1,97 @@
+            "prefix_suffix": ["|", "|"]}
+    assert utils.string_split(**kwds) == ['|', 'p', 'y', ' ', '0', '0', '1', ' ', '$', ' ', ' ', '|']
+    
+    # -------------------
+    kwds = {"x": "py 001 $  ", 
+            "tokenize": ["char"],  
+            "prefix_suffix": ["|", None]}
+    assert utils.string_split(**kwds) == ['p', 'y', ' ', '0', '0', '1', ' ', '$', ' ', ' ']
+    
+    # -------------------
+    kwds = {"x": "py 001 $  ", 
+            "tokenize": ["char"], 
+            "prefix_suffix": [None]}
+    assert utils.string_split(**kwds) == ['p', 'y', ' ', '0', '0', '1', ' ', '$', ' ', ' ']
+    
+    # -------------------
+    kwds = {"x": "py 001 $  ", 
+            "tokenize": ["char"], 
+            "prefix_suffix": None}
+    assert utils.string_split(**kwds) == ['p', 'y', ' ', '0', '0', '1', ' ', '$', ' ', ' ']
+    
+    # -------------------
+    # min_gram must be >= 1
+    kwds = {"x": "py 001 $  ", 
+            "tokenize": ["char", "ngram", "word"], 
+            "min_gram": 0, 
+            "max_gram": 3, 
+            "prefix_suffix": ["|", "|"]}
+    with pytest.raises(AssertionError):
+        utils.string_split(**kwds)
+    
+    # -------------------
+    # max_gram must be >= min_gram
+    kwds = {"x": "py 001 $  ",  
+            "tokenize": ["char", "ngram", "word"], 
+            "min_gram": 3, 
+            "max_gram": 2,  
+            "prefix_suffix": ["|", "|"]}
+    with pytest.raises(AssertionError):
+        utils.string_split(**kwds)
+    
+    # -------------------
+    kwds = {"x": "py 001 $  ",  
+            "tokenize": ["ngram"], 
+            "min_gram": 1, 
+            "max_gram": 1, 
+            "prefix_suffix": None}
+    assert utils.string_split(**kwds) == ['p', 'y', ' ', '0', '0', '1', ' ', '$', ' ', ' ']
+    
+    # -------------------
+    kwds = {"x": "py 001 $  ", 
+            "tokenize": ["ngram", "word"], 
+            "min_gram": 3, 
+            "max_gram": 3, 
+            "token_sep": "$", 
+            "prefix_suffix": None}
+    assert utils.string_split(**kwds) == ['py ', 'y 0', ' 00', '001', '01 ', '1 $', ' $ ', '$  ', 'py 001 ', '  ']
+    
+    # -------------------
+    kwds = {"x": "py 001 $  ", 
+            "tokenize": ["ngram", "word"], 
+            "min_gram": 3, 
+            "max_gram": 3, 
+            "token_sep": "$", 
+            "prefix_suffix": ["|", "|"]}
+    assert utils.string_split(**kwds) == ['|py', 'py ', 'y 0', ' 00', '001', '01 ', '1 $', ' $ ', '$  ', '  |', 'py 001 ', '  ']
+
+    # -------------------
+    kwds = {"x": "py 001 $  ", 
+            "tokenize": ["word"], 
+            "token_sep": "default", 
+            "prefix_suffix": None}
+    assert utils.string_split(**kwds) == ['py', '001']
+
+    # -------------------
+    kwds = {"x": "py 001 $  ", 
+            "tokenize": ["word"], 
+            "token_sep": "$", 
+            "prefix_suffix": None}
+    assert utils.string_split(**kwds) == ['py 001 ', '  ']
```

Comment:
@kasra-hosseini, should we also add a case in which there is more than one token separator? For example (I added the white space as a delimiter):
```
    kwds = {"x": "py 001 $  ", 
            "tokenize": ["word"], 
            "token_sep": "$ ", 
            "prefix_suffix": None}
    assert utils.string_split(**kwds) == ['py', '001']
```

Reply:
Nice! Sure... Done:https://github.com/Living-with-machines/DeezyMatch/pull/111/commits/7ba49d6914993d4628cfbecd3d9d09bb62bac444